cli/.js/src/main/scala/aqua/ArgOpts.scala

```diff
@@ -50,8 +50,34 @@ object ArgOpts {
                 validNel(LiteralRaw(value, ts))
               case VarToken(name, _) =>
                 validNel(VarRaw(name.value, BottomType))
-              case CollectionToken(_, _) =>
-                invalidNel("Array argument in function call not supported. Pass it through JSON.")
+              case CollectionToken(_, values) =>
+                val hasVariables = values.exists {
+                  case LiteralToken(_, _) => false
+                  case _ => true
+                }
+                if (!hasVariables) {
+                  val literals = values.collect { case LiteralToken(value, ts) =>
+                    LiteralRaw(value, ts)
+                  }
+                  val hasSameTypesOrEmpty =
+                    literals.isEmpty || literals.map(_.baseType).toSet.size == 1
+
+                  if (hasSameTypesOrEmpty) {
+                    validNel(
+                      NonEmptyList
+                        .fromList(literals)
+                        .map(l => CollectionRaw(l, ArrayType(l.head.baseType)))
+                        .getOrElse(ValueRaw.Nil)
+                    )
+                  } else
+                    invalidNel(
+                      "If the argument is an array, then it must contain elements of the same type."
+                    )
+
+                } else
+                  invalidNel(
+                    "Array arguments with variables are not supported. Use only numbers, strings, or booleans."
```

Comment:
We should make sure that we validate that array contains only booleans, strings or numbers and the error message should simply be "Arrays can only have numbers, strings, or booleans." or something like this